File: components_page/components/dropdown/direction.py
import dash_html_components as html
import dash_bootstrap_components as dbc

items = [
    dbc.DropdownMenuItem("First"),
    dbc.DropdownMenuItem(divider=True),
    dbc.DropdownMenuItem("Second"),
]

dropdown = html.Div(
    [
        html.H4("Dropdown direction"),
        dbc.Row(
            [
                dbc.Col(
                    dbc.DropdownMenu(
                        label="Dropup", children=items, direction="up"
                    )
                ),
                dbc.Col(
                    dbc.DropdownMenu(
                        label="Dropleft", children=items, direction="left"
                    )
                ),
                dbc.Col(
                    dbc.DropdownMenu(
                        label="Dropright", children=items, direction="right"
                    )
                ),
            ]
        ),
    ]
)

File: components_page/components/dropdown/direction.R
library(dashBootstrapComponents)

items_direction <- list(
  dbcDropdownMenuItem("First"),
  dbcDropdownMenuItem(divider = TRUE),
  dbcDropdownMenuItem("Second")
)

dropdown <- dbcRow(
  list(
    dbcCol(
      dbcDropdownMenu(
        label = "Dropdown (default)", children = items_direction,
        direction = "down"
      ),
      width = "auto"
    ),
    dbcCol(
      dbcDropdownMenu(
        label = "Dropleft", children = items_direction, direction = "left"
      ),
      width = "auto"
    ),
    dbcCol(
      dbcDropdownMenu(
        label = "Dropright", children = items_direction, direction = "right"
      ),
      width = "auto"
    ),
    dbcCol(
      dbcDropdownMenu(
        label = "Dropup", children = items_direction,
        direction = "up"
      ),
      width = "auto"
    )
  ),
  justify = "between"
)
